packages/driver/src/multi-domain/communicator.ts

```diff
@@ -117,6 +79,11 @@ export class PrimaryDomainCommunicator extends EventEmitter {
   }
 
   toSpecBridge (domain: string, event: string, data?: any) {
+    if (data?.userInvocationStack) {
+      this.userInvocationStack = data.userInvocationStack
+      delete data.userInvocationStack
+    }
```

Comment:
Instead of attaching this as a byproduct of an event, why don't we attach it to the communicator explicitly. `PrimaryDomainCommunicator .setUserInvocationStack`

I think that would be a lot more clear.

Reply:
@mjhenkes are you thinking calling `PrimaryDomainCommunicator.setUserInvocationStack` at the beginning of the `switchToDomain` call? Definitely would be a lot clearer as to when this happens and simpler

Reply:
@AtofStryker yep, exactly!

Reply:
I wonder if it makes sense to just set it on the instance directly, like `Cypress.multiDomainCommunicator.userInvocationStack = `state('current').get('userInvocationStack')`. I feel like this is super clear but wanted to get your thoughts

Reply:
@AtofStryker I think that is fine too